File: banda/SergioDeIscar-banda-java/SergioDeIscar-banda-java/src/main/java/org/example/models/CantanteGuitarrista.java
package org.example.models;

import org.example.enums.GuitarraType;
import org.example.enums.TonoType;
import org.example.interfaces.ICantante;
import org.example.interfaces.IGuitarrista;

public final class CantanteGuitarrista extends Musician implements ICantante, IGuitarrista {
    private TonoType tonoType;
    private GuitarraType guitarraType;
    private final Float salario = super.salario * 1.5f;
    public CantanteGuitarrista(String nombre, int experiencia, TonoType tono, GuitarraType guitarra) {
        super(nombre, experiencia);
        setTono(tono);
        setGuitarra(guitarra);
    }

    @Override
    public TonoType getTono() {
        return tonoType;
    }

    @Override
    public void setTono(TonoType tono) {
        tonoType = tono;
    }

    @Override
    public void cantar() {
        System.out.println("El cantanteGuitarrista " + nombre + " esta cantando");
    }

    @Override
    public GuitarraType getGuitarra() {
        return guitarraType;
    }

    @Override
    public void setGuitarra(GuitarraType guitarra) {
        guitarraType = guitarra;
    }

    @Override
    public void guitarrear() {
        System.out.println("El cantanteGuitarrista " + nombre + " esta guitarreando");
    }

    @Override
    public float getSalario() {
        return salario;
    }

    @Override
    public String getNombre() {
        return this.nombre;
    }
}

File: banda/SergioDeIscar-banda-kotlin/SergioDeIscar-banda-kotlin/src/main/kotlin/models/CantanteGuitarrista.kt
package bandamusical.models

import bandamusical.interfeces.ICantante
import bandamusical.interfeces.IGuitarrista

class CantanteGuitarrista(nombre:String,
                          override val experiencia: Int,
                          override val tono: TonoType,
                          override val guitarra: GuitarraType): Musician(nombre), ICantante, IGuitarrista {
    override val salario: Float = super.salario * 1.5f
    override fun cantar() {
        println("El cantanteGuitarrista $nombre canta con un tono $tono.")
    }

    override fun guitarrear() {
        println("El cantanteGuitarrista $nombre gitarrea, el tipo de gitarra utilizado es $guitarra.")
    }
}